### ISSUE TITLE
BUG: np.average unable to specify axes

### ISSUE THREAD

**Issue by baloe** [numpy/numpy#27040]
### Describe the issue:

I am unable to use `np.average` with a tuple of axes specified.
I can reproduce the error when running the following example which I copied from the documentation
https://numpy.org/doc/stable/reference/generated/numpy.average.html

### Reproduce the code example:

```python
import numpy as np
data = np.arange(8).reshape((2, 2, 2))
np.average(data, axis=(0, 1), weights=[[1./4, 3./4], [1., 1./2]])
```

### Error message:

```shell
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/soft/mambaforge/envs/condaforge/lib/python3.11/site-packages/numpy/lib/function_base.py", line 538, in average
    raise TypeError(
TypeError: 1D weights expected when shapes of a and weights differ.
```

### Python and NumPy Versions:

1.26.4
3.11.9 | packaged by conda-forge | (main, Apr 19 2024, 18:36:13) [GCC 12.3.0]

### Runtime Environment:

```
[{'numpy_version': '1.26.4',
  'python': '3.11.9 | packaged by conda-forge | (main, Apr 19 2024, 18:36:13) '
            '[GCC 12.3.0]',
  'uname': uname_result(system='Linux', node='mtebenberg', release='5.15.0-116-generic', version='#126~20.04.1-Ubuntu SMP Mon Jul 1 15:40:07 UTC 2024', machine='x86_64')},
 {'simd_extensions': {'baseline': ['SSE', 'SSE2', 'SSE3'],
                      'found': ['SSSE3',
                                'SSE41',
                                'POPCNT',
                                'SSE42',
                                'AVX',
                                'F16C',
                                'FMA3',
                                'AVX2'],
                      'not_found': ['AVX512F',
                                    'AVX512CD',
                                    'AVX512_KNL',
                                    'AVX512_KNM',
                                    'AVX512_SKX',
                                    'AVX512_CLX',
                                    'AVX512_CNL',
                                    'AVX512_ICL',
                                    'AVX512_SPR']}},
 {'architecture': 'Haswell',
  'filepath': '/soft/mambaforge/envs/condaforge/lib/libopenblasp-r0.3.27.so',
  'internal_api': 'openblas',
  'num_threads': 8,
  'prefix': 'libopenblas',
  'threading_layer': 'pthreads',
  'user_api': 'blas',
  'version': '0.3.27'}]
```

**Comment by ngoldbaum** [comment]
I can reproduce this on NumPy 1.26.4, but it's fixed on `main`:

```
Preimported NumPy 2.1.0.dev0+git20240725.2d7ede4 as np

In [1]: import numpy as np
   ...: data = np.arange(8).reshape((2, 2, 2))
   ...: np.average(data, axis=(0, 1), weights=[[1./4, 3./4], [1., 1./2]])
Out[1]: array([3.4, 4.4])
```

**Comment by ngoldbaum** [comment]
I don't think we're backporting to the 1.26 branch anymore.

**Comment by charris** [comment]
> I don't think we're backporting to the 1.26 branch anymore.

Correct. But we may do another 2.0.x  release, there are now two bugs fixed, and three would push the issue.

**Comment by ngoldbaum** [comment]
I didn't try with 2.0.1, is it fixed on the 2.0 branch?

**Comment by WarrenWeckesser** [comment]
The example works with 2.0.0:
```
In [6]: import numpy as np

In [7]: np.__version__
Out[7]: '2.0.0'

In [8]: data = np.arange(8).reshape((2, 2, 2))

In [9]: np.average(data, axis=(0, 1), weights=[[1/4, 3/4], [1, 1/2]])
Out[9]: array([3.4, 4.4])
```

**Comment by WarrenWeckesser** [comment]
A work-around for 1.26.4 is to do some careful reshaping and use an integer axis.  For example,
```
In [8]: data = np.arange(8).reshape((2, 2, 2))

In [9]: weights = np.array([[1/4, 3/4], [1, 1/2]])

In [10]: np.average(data.reshape(4, 2), axis=0, weights=weights.ravel())
Out[10]: array([3.4, 4.4])
```

Closing, since this is fixed in 2.0.0, and there are no more backports for 1.26 planned.
